### DISCUSSION TITLE
хочу на кнопку повесить действие кода для api telegram чтоб вытащить параметры из таблицы

### DISCUSSION THREAD

**Discussion by IrinaG1090** [totumonline/totum-ru-issues-and-discussions#446] [Как это сделать?] [answered]
Народ, привет! помогите, где посмотреть - какие параметры передавать ?
Есть таблица цикла с полями, поля в цикле тип Ярлык -ссылка. Связь идет через ярлык-ссылку с полями из расчетной в цикле. (поле1 числовое в расчетной, в цикле поле1-ярлык, поле2 селект в расчетной, в цикле поле2-ярлык.)
Передать нужно поле1 и поле2 из цикла в телеграм. То есть поле1 (тип данных ярлык-ссылка) и поле2 (тип данных ярлык-ссылка). Делаю это через кнопку. В ней пишу код действия, как в видеоуроке https://ru.docs.totum.online/telegram-bot. 
Как написать обработчик правильно? чтоб в телегу приходили поле1 и поле2 из выбранной таблицы.

вот такой код :

```
=: linkToDataJson(title: "Ответ"; data: $extr)

extr: jsonExtract(text: $getfs)

getfs: getFromScript(uri: str`"https://api.telegram.org/bot" + $token + "/" + $command`; post: "chat_id" = "***"; **post: "text" = $text)
//что добавить в конце строки кода, какие параметры? прописывать ?

token: "***"
command: "sendMessage"

text: selectRow(table: 'tablename'; field: 'pole1'; field: 'pole2'; field: 'pole3') 
//=> вот это нужно обработать. ума не приложу через какие параметры это делается, решила достать это через selectRow, но понимаю что обработчик не знает какие поля брать-какие передавать и каким методом. и как эти поля связать. знаю что это должно быть вида значение1=значение2. но как реализовать - не пойму. измучалась уже.

//text: "ЛА-Ла-Ла!!! Посмотреть можно в личном кабинете http://totum.ru/Auth/Login/?from=%2F" `  
//=> сейчас стоит вот такая заглушка

```

![форма](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/23610956/61505c98-7ed7-4d24-b297-f035709db2bc)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Я рекомендую вам пересоздать бота тк вы выложили токен от него в открытый доступ и это осталось в истории интернета (если это ваш реальный токен).

В text должен быть текст:

```
text: str`"Продажи в" ++ $row[months] ++ "составили" ++ $row[sum]`

~row: selectRow(table: 'tablename'; field: 'months'; field: 'sum'; where: '...' = ...)

```

**Reply by IrinaG1090** [reply]
Благодарю, попробую -))
